homeassistant/components/edl21/sensor.py

```diff
@@ -36,6 +41,203 @@
     },
 )
 
+# OBIS format: A-B:C.D.E*F
+SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
+    # A=1: Electricity
+    # C=0: General purpose objects
+    # D=0: Free ID-numbers for utilities
+    SensorEntityDescription(
+        key="1-0:0.0.9*255", name="Electricity ID", icon="mdi:flash"
+    ),
+    # D=2: Program entries
+    SensorEntityDescription(
+        key="1-0:0.2.0*0", name="Configuration program version number", icon="mdi:flash"
+    ),
+    SensorEntityDescription(
+        key="1-0:0.2.0*1", name="Firmware version number", icon="mdi:flash"
+    ),
+    # C=1: Active power +
+    # D=8: Time integral 1
+    # E=0: Total
+    SensorEntityDescription(
+        key="1-0:1.8.0*255",
+        name="Positive active energy total",
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_class=SensorDeviceClass.ENERGY,
+    ),
+    # E=1: Rate 1
+    SensorEntityDescription(
+        key="1-0:1.8.1*255",
+        name="Positive active energy in tariff T1",
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_class=SensorDeviceClass.ENERGY,
```

Comment:
I'm missing an explicit native unit of measurement set for the entity descriptions with a device class. We only allow specific units when using our device classes. We have constants for these units in const.py that should be used.

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes